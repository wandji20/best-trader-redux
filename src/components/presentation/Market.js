import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FaArrowUp, FaArrowDown } from 'react-icons/fa';

const Market = (props) => {
  const { market } = props;
  const {
    ticker, bid, ask, changes, high, low,
  } = market;
  // const params = ticker.split('/').join('');
  return (
    <article id="article" className="col-sm-6 col-md-4 d-flex flex-column bg-dark text-white">
      <h5>{ticker}</h5>
      <div className=" d-flex flex-column">
        <p className="d-flex justify-content-between">
          <span>
            Bid:
            {bid}
          </span>
          <span>
            Ask:
            {ask}
          </span>
        </p>
        <p className="d-flex justify-content-center">
          {
            changes > 0
              ? (
                <span className="text-info">
                  <FaArrowUp />
                  {' '}
                  {changes}
                </span>
              )
              : (
                <span className="text-danger">
                  <FaArrowDown />
                  {' '}
                  {changes}
                </span>
              )
          }

        </p>
        <p className="d-flex justify-content-between">
          <span>
            High:
            {high}
          </span>
          <span>
            Low:
            {low}
          </span>
        </p>
        <p className="d-flex justify-content-between">
          <button type="button" className="btn btn-success">BUY</button>
          <button type="button" className="btn btn-danger">SELL</button>
          <Link
            to={
                {
                  pathname: '/Details',
                  hash: ticker,
                  state: { ticker },
                }
              }
          >
            <button type="button" className="btn btn-info">MORE</button>
          </Link>
        </p>
      </div>
    </article>
  );
};

Market.propTypes = {
  market: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default Market;
