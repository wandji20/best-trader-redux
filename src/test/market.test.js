import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Market from '../components/presentation/Market';

const market = {
  ticker: 'CAD/JPY',
  bid: '88.478',
  ask: '88.478',
  open: '87.582',
  low: '87.442',
  high: '88.486',
  changes: '1.0230412641867062',
  date: '2021-07-09 16:01:46',
};

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Market market={market} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
