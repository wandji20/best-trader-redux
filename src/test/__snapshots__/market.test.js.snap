// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="col-sm-6 col-md-4 d-flex flex-column bg-dark text-white"
  id="article"
>
  <h5>
    CAD/JPY
  </h5>
  <div
    className=" d-flex flex-column"
  >
    <p
      className="d-flex justify-content-between"
    >
      <span>
        Bid:
         
        88.478
      </span>
      <span>
        Ask:
         
        88.478
      </span>
    </p>
    <p
      className="d-flex justify-content-center"
    >
      <span
        className="text-info"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
          />
        </svg>
         
        1.023
      </span>
    </p>
    <p
      className="d-flex justify-content-between"
    >
      <span>
        High:
         
        88.486
      </span>
      <span>
        Low:
         
        87.442
      </span>
    </p>
    <p
      className="d-flex justify-content-between"
    >
      <button
        className="btn btn-success"
        type="button"
      >
        BUY
      </button>
      <button
        className="btn btn-danger"
        type="button"
      >
        SELL
      </button>
      <a
        href="/Details#CAD/JPY"
        onClick={[Function]}
      >
        <button
          className="btn btn-info"
          type="button"
        >
          MORE
        </button>
      </a>
    </p>
  </div>
</article>
`;
